### ISSUE TITLE
computeColumnDifference reports false differences for columns with a default on Postgres

### ISSUE THREAD

**Issue by thekevinbrown** [mikro-orm/mikro-orm#380]
**Describe the bug**

When I use

```typescript
const generator = orm.getSchemaGenerator();
const dump = await generator.getUpdateSchemaSQL(false);
```

while not having actually changed anything in my database since the last time I ran `getUpdateSchemaSQL`, I get something like the following on a field that has a default value:

```typescript
await database.query('alter table "organisation" alter column "retrieval_left" drop default;');
await database.query('alter table "organisation" alter column "retrieval_left" drop not null;');
await database.query('alter table "organisation" alter column "retrieval_left" type int4 using ("retrieval_left"::int4);');
await database.query('alter table "organisation" alter column "retrieval_left" set default -1;');
await database.query('alter table "organisation" alter column "retrieval_left" set not null;');
```

It looks like `computeColumnDifference` doesn't notice that defaults are already set and shouldn't be dropped then re-created.

**Stack trace**

No crash, I think this is coming from the comparison here:

https://github.com/mikro-orm/mikro-orm/blob/master/lib/schema/SchemaHelper.ts#L44

**To Reproduce**

1) Create an entity:

```typescript
@Entity()
export class Organisation {
	@PrimaryKey({ type: 'number' })
	id!: number;

	@Property({ type: 'number', nullable: false, default: -1 })
	retrievalLeft!: number;
}
```

2) Run

```typescript
const generator = orm.getSchemaGenerator();
const dump = await generator.getUpdateSchemaSQL(false);
```

3) Run the statements in `dump` against the local Postgres database
4) Run the code in step 2 again. You should get something like:

```typescript
await database.query('alter table "organisation" alter column "retrieval_left" drop default;');
await database.query('alter table "organisation" alter column "retrieval_left" drop not null;');
await database.query('alter table "organisation" alter column "retrieval_left" type int4 using ("retrieval_left"::int4);');
await database.query('alter table "organisation" alter column "retrieval_left" set default -1;');
await database.query('alter table "organisation" alter column "retrieval_left" set not null;');
```

**Expected behavior**

Since the creation migration ran, I expected the schema generator to see that the default is already set on the column.

**Versions**

| Dependency | Version |
| - | - |
| node | 10.18.1 |
| typescript | 3.8.2 |
| mikro-orm | 3.3.4  |
| your-driver | pg 7.18.2 |

**Comment by B4nan** [comment]
Cannot replicate this, following test is passing for me:

```typescript
import { unlinkSync } from 'fs';
import { Entity, PrimaryKey, Property, MikroORM, ReflectMetadataProvider } from '../../lib';
import { SqliteDriver } from '../../lib/drivers/SqliteDriver';

@Entity()
class A {

  @PrimaryKey()
  id!: number;

  @Property({ default: -1 })
  foo!: number;

}

describe('GH issue 380', () => {

  let orm: MikroORM<SqliteDriver>;

  beforeAll(async () => {
    orm = await MikroORM.init({
      entities: [A],
      dbName: __dirname + '/../../temp/mikro_orm_test_gh380.db',
      debug: false,
      highlight: false,
      type: 'sqlite',
      metadataProvider: ReflectMetadataProvider,
      cache: { enabled: false },
    });
    await orm.getSchemaGenerator().dropSchema();
    await orm.getSchemaGenerator().createSchema();
  });

  afterAll(async () => {
    await orm.close(true);
    unlinkSync(orm.config.get('dbName'));
  });

  test(`schema updates respect default values`, async () => {
    const generator = orm.getSchemaGenerator();
    const dump = await generator.getUpdateSchemaSQL(false);
    expect(dump).toBe('');
  });

});
```

**Comment by B4nan** [comment]
Maybe its only a postgres issue, I suspect the column type to be a problem, not the default value.

**Comment by B4nan** [comment]
Fixed in v3.3.5

**Comment by thekevinbrown** [comment]
Thanks @B4nan, I can confirm that with v3.3.5 this is fixed. Much appreciated!
